scri/flux.py

```diff
@@ -437,12 +429,298 @@ def angular_momentum_flux(h, hdot=None):
+
+
+def boost_flux(h, hdot=None):
+    """p_z, p_plusminus and the matrix elements defined above will be used here.
+    matrix_expectation_value is used to calcuate the component of the flux.
+    The spin value taken by the matrix element is specified for each term.
+    """
+
+    from .waveform_modes import WaveformModes
+    from . import h as htype
+    from . import hdot as hdottype
+
+    if not isinstance(h, WaveformModes):
+        raise ValueError(
+            "Boost fluxes can only be calculated from a `WaveformModes` object; "
+            + "`h` is of type `{0}`.".format(type(h))
+        )
+    if (hdot is not None) and (not isinstance(hdot, WaveformModes)):
+        raise ValueError(
+            "Boost fluxes can only be calculated from a `WaveformModes` object; "
+            + "`hdot` is of type `{0}`.".format(type(hdot))
+        )
+    if h.dataType == htype:
+        if hdot is None:
+            hdot = h.copy()
+            hdot.dataType = hdottype
+            hdot.data = h.data_dot
+        elif hdot.dataType != hdottype:
+            raise ValueError(
+                "Input argument `hdot` is expected to have data of type `hdot`; "
+                + "this `hdot` waveform data has type `{0}`".format(h.data_type_string)
+            )
+    else:
+        raise ValueError(
+            "Input argument `h` is expected to have data of type `h`; "
+            + "this `h` waveform data has type `{0}`".format(h.data_type_string)
+        )
+
+    # h and h_dot are spin raised and lowered in the expression for boost flux.
+    # Hence, one needs to define new waveform like object from h and hdot that
+    # contains the updated data corresponding to the raising and lowering of the spin.
+    # Appropriate ladder factor needs to be multiplied to each (l,m) mode.
+    # eth and ethbar functions are used here for this operation.
+
+    # eth_h --> Spin of h raised by 1; s = -1
+    eth_h = h.copy()
+    eth_h.data = h.eth
+
+    # ethbar_h --> Spin of h lowered by 1; s = -3
+    ethbar_h = h.copy()
+    ethbar_h.data = h.ethbar
+
+    # eth_hdot --> Spin of h raised by 1; s = -1
+    eth_hdot = hdot.copy()
+    eth_hdot.data = hdot.eth
+
+    # ethbar_hdot --> Spin of h lowered by 1; s = -3
+    ethbar_hdot = hdot.copy()
+    ethbar_hdot.data = hdot.ethbar
+
+    boost_flux = np.zeros((hdot.n_times, 3), dtype=float)
+
+    # Each term has been computed one by taking expectation value
+    # between waveform like objects (h, hdot, eth_h, ethbar_h, eth_hdot, ethbar_hdot)
+    # and the matrix elements.
+    # The spin of the waveform like object is taken as input for spin value s in matrix elements p_z and p_plusminus.
+    # Because both <a| and |b> have the same spin when the matrix element is p_z and p_plusminus.
+
+    # plus and minus terms are computed here.
+    _, ethbar_hdot_chiplus_ethbar_h = matrix_expectation_value(ethbar_hdot, functools.partial(p_plus, s=-3), ethbar_h)
```

Comment:
Optional:

`matrix_expectation_value` returns two objects and you only want the second object. You can do what you are doing here but you can also do:

```
ethbar_hdot_chiplus_ethbar_h = matrix_expectation_value(ethbar_hdot, functools.partial(p_plus, s=-3), ethbar_h)[1]
```
It's up to you.

Reply:
Then, I will have to add the [1] in front of all the matr_ex_val. Won't it make it unattractive?